magicctapipe/scripts/lst1_magic/lst1_magic_cloud_correction.py

```diff
@@ -0,0 +1,715 @@
+        obs_id = dl1_params["obs_id"][index]
+        event_id_magic = dl1_params["event_id_magic"][index]
+        obs_id_magic = dl1_params["obs_id_magic"][index]
+        timestamp = dl1_params["timestamp"][index]
+        multiplicity = dl1_params["multiplicity"][index]
+        combo_type = dl1_params["combo_type"][index]
+
+        if tel_ids["LST-1"] == tel_id:
+            event_id_image, obs_id_image = event_id_lst, obs_id_lst
+        else:
+            event_id_image, obs_id_image = event_id_magic, obs_id_magic
+
+        pointing_az = dl1_params["pointing_az"][index]
+        pointing_alt = dl1_params["pointing_alt"][index]
+        time_diff = dl1_params["time_diff"][index]
+        n_islands = dl1_params["n_islands"][index]
+        signal_pixels = dl1_params["n_pixels"][index]
+
+        alt_rad = np.deg2rad(dl1_params["alt"][index])
+        az_rad = np.deg2rad(dl1_params["az"][index])
+
+        impact = dl1_params["impact"][index] * u.m
+        cog_x = (dl1_params["x"][index] * m2deg).value * u.deg
+        cog_y = (dl1_params["y"][index] * m2deg).value * u.deg
+
+        # Source position
+        reco_pos = SkyCoord(alt=alt_rad * u.rad, az=az_rad * u.rad, frame=AltAz())
+        telescope_pointing = SkyCoord(
+            alt=pointing_alt * u.rad,
+            az=pointing_az * u.rad,
+            frame=AltAz(),
+        )
+
+        tel_frame = TelescopeFrame(telescope_pointing=telescope_pointing)
+        tel = reco_pos.transform_to(tel_frame)
+
+        src_x = tel.fov_lat
+        src_y = tel.fov_lon
+
+        # Transform to Engineering camera
+        src_x, src_y = -src_y, -src_x
+        cog_x, cog_y = -cog_y, -cog_x
+
+        psi = np.arctan2(src_x - cog_x, src_y - cog_y)
+
+        pix_x_tel = (camgeom.pix_x * m2deg).to(u.deg)
+        pix_y_tel = (camgeom.pix_y * m2deg).to(u.deg)
+
+        distance = np.abs(
+            (pix_y_tel - src_y) * np.cos(psi) + (pix_x_tel - src_x) * np.sin(psi)
+        )
+
+        d2_cog_src = (cog_x - src_x) ** 2 + (cog_y - src_y) ** 2
+        d2_cog_pix = (cog_x - pix_x_tel) ** 2 + (cog_y - pix_y_tel) ** 2
+        d2_src_pix = (src_x - pix_x_tel) ** 2 + (src_y - pix_y_tel) ** 2
+
+        distance[d2_cog_pix > d2_cog_src + d2_src_pix] = 0
+        dist_corr_layer = model2(impact, Hcl, mean_subrun_zenith) * u.deg
+
+        ilayer = np.digitize(distance, dist_corr_layer)
+        trans_pixels = transl[ilayer]
+
+        inds_img = np.where(
+            (dl1_images["event_id"] == event_id_image)
+            & (dl1_images["tel_id"] == tel_id)
+            & (dl1_images["obs_id"] == obs_id_image)
+        )[0]
+
+        if len(inds_img) == 0:
+            raise ValueError("Error: 'inds_img' list is empty!")
+        index_img = inds_img[0]
+        image = dl1_images["image"][index_img]
+        clean_mask = dl1_images["image_mask"][index_img]
+        peak_time = dl1_images["peak_time"][index_img]
+        image /= trans_pixels
+
+        clean_image = image[clean_mask]
```

Comment:
you use the old mask first no matter if you do the additional cleaning a moment later or not. I think it is faster to move those 3 lines as "else" after the next condition